obj.go

```diff
@@ -217,6 +217,34 @@ func (O *Object) AsMap(recursive bool) (map[string]interface{}, error) {
 	return m, nil
 }
 
+func (o *Object) ObjAsMap() (map[string]interface{}, error) {
+	m := make(map[string]interface{}, len(o.ObjectType.Attributes))
+	for a := range o.ObjectType.Attributes {
+		var err error
+		if m[a], err = o.Get(a); err != nil {
+			return m, fmt.Errorf("%q: %w", a, err)
+		}
+		if sub, ok := m[a].(*Object); ok {
+			if m[a], err = sub.ObjAsMap(); err != nil {
+				return m, fmt.Errorf("%q sub: %w", a, err)
+			}
+		} else if subs, ok := m[a].(*ObjectCollection); ok {
+			if r, err := subs.CollAsArrayMap(); err != nil {
+				return m, fmt.Errorf("%q: %w", a, err)
+			} else {
+				m[a] = r
+			}
+		}
+	}
+	return m, nil
+}
+
+func (o *Object) GetFromMap(m map[string]interface{}) {
```

Comment:
This horribly inefficient convenience function (just a for loop) should not be in the library.

Reply:
it goes with GetFromArrayMap, to give the same option for the obj and for the collection.
I think it can be fine for converting the value then to a json (and i need it too, so taking into the library is good. Even i'll write it into my code).

Another point will be to convert to a scan, but i haven't figured it out yet.